lib/storage.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const Promise = require('bluebird');
+const fs = Promise.promisifyAll(require('fs'), {suffix: 'Prom'});
+
+module.exports = exports = {};
+
+exports.createItem = function(schemaName, item) {
+  if(!schemaName) return Promise.reject(new Error('expected schema name'));
+  if(!item) return Promise.reject(new Error('expected item'));
+
+  let json = JSON.stringify(item);
+  return fs.writeFileProm(`${__dirname}/../data/${schemaName}/${item.id}.json`, json)
+  .then( () => item)
+  .catch( err => Promise.reject(err));
+};
+
+exports.fetchItem = function(schemaName, id) {
+  if(!schemaName) return Promise.reject(new Error('expected schema name'));
+  if(!id) return Promise.reject(new Error('expected id'));
+
+  return fs.readFileProm(`${__dirname}/../data/${schemaName}/${id}.json`)
+  .then(data => {
+    try {
+      let item = JSON.parse(data.toString());
+      return item;
+    } catch (err) {
+      return Promise.reject(err);
+    }
+  })
+  .catch( err => Promise.reject(err));
+};
```

Comment:
You are missing a delete route. In order to make a delete route work you need to add a method here (i.e. `exports.deleteItem`) that uses `fs.unlink()`.